### ISSUE TITLE
Improve information about the cause of failure during hub health check

### ISSUE THREAD

**Issue by sgibson91** [2i2c-org/infrastructure#1206]
### Context

There are different reasons why the hub health check might fail:

- The notebook didn't execute correctly
- Kubernetes-related issues with provisioning the pod and/or node during the check
- Other reasons I haven't considered

This information is helpful to engineers to reason about a check failure and make a decision like "is it ok to just re-run the check or should I investigate more deeply?"

### Proposal

To reduce ambiguity around check failures related to Kubernetes processes, we could report the following during the check:

- Was the pod scheduled successfully?
- Have the pod's containers started successfully?
- Did the pod's containers finish without error?

Is it possible that this is information `jhub-client` is already aware of that we could propagate through to our check logs?

**Comment by consideRatio** [comment]
I agree on the helpfulness about this. I'm especially interested in understanding if the test fails because image pulling took too long as some images can be huge, and sometimes the networking speed can be reduced etc.

I think #1232 to stop using pytest should be done as the first step towards this
